.github/scripts/mr_publish_results.py

```diff
@@ -238,6 +224,17 @@ def send_to_segment(context: Dict[str, Any]):
     )
 
 
+def _get_is_external_and_dataset_repository_branch():
+    is_external = os.environ["IS_EXTERNAL"]
+    dataset_repository_branch = os.environ["DATASET_REPOSITORY_BRANCH"]
+    if str(is_external).lower() in ("yes", "true", "t", "1"):
```

Comment:
environment variables are always strings, aren't they?
```suggestion
    if is_external.lower() in ("yes", "true", "t", "1"):
```